File: Assets/CityGame/Lua/Dbg.lua

Dbg = {}

local this = Dbg;

--Output log--
local loginner = print
DEBUGLEVEL = {
    DEBUG = 0,
    INFO = 1,
    WARNING = 2,
    ERROR = 3,
    NOLOG = 4,  -- Put it at the end, when using this, it means that no logs will be output (!!!Use with caution!!!)
}

if ct.G_DEBUGLOG then
    this.debugLevel = DEBUGLEVEL.DEBUG;
else
    this.debugLevel = DEBUGLEVEL.NOLOG
end

this.getHead = function()
    return "";
end
this.getIdHead = function(id)
    return "["..id.."]";
end

log = nil

--ct.log = function(logid,s,...)
--	if s == nil then
--		return
--	end
--	local plgid = TestGroup.get_TestGroupId(logid)
--	local palgid = TestGroup.get_ActiveTestGroupId(logid)
--	if plgid == nil or  palgid == nil then
--		return
--	end
--	assert(s)
--	loginner(this.getIdHead(plgid) .. s,...);
--end

--print = function(s,...)
--	ct.log("system",s,...)
--end

logWarn = print
logDebug = print

error = function(s,...)
    ct.log("system","[error]",s,...)
end

this.INFO_MSG = function( s )
    if (DEBUGLEVEL.INFO >= this.debugLevel) then
        ct.log(this.getHead() .. s);
    end
end

this.DEBUG_MSG = function( s ,...)
    if (DEBUGLEVEL.DEBUG >= this.debugLevel) then
        ct.log(this.getHead() .. s,...);
    end
end

this.WARNING_MSG = function(s)
    if (DEBUGLEVEL.WARNING >= this.debugLevel) then
        CityLuaUtil.LogWarning(this.getHead() .. s);
    end
end

this.ERROR_MSG = function(s)
    if (DEBUGLEVEL.ERROR >= this.debugLevel) then
        CityLuaUtil.LogError(this.getHead() .. s);
    end
end

--debug output log--
function logDebug(str,...)
    assert(str,"logdebug nil")
    Dbg.DEBUG_MSG(str,...);
end

--Error log--
function logError(str)
    assert(str,"logError nil")
    Dbg.ERROR_MSG(str);
end

--Warning log--
function logWarn(str)
    assert(str,"logWarn nil")
    Dbg.WARNING_MSG(str);
end

File: Assets/CityGame/Scripts/Framework/Dbg.cs
using UnityEngine;
using City;
using System;
using System.Collections;
using System.Collections.Generic;

namespace City
{
	public enum DEBUGLEVEL : int
	{
		DEBUG = 0,
		INFO,
		WARNING,
		ERROR,

		NOLOG,  // Put it at the end, when using this, it means that no logs will be output (!!!Use with caution!!!)
	}

    public enum SYSEVENT : int
    {
        SYSEVENT_DEFAULT = 0,
        SYSEVENT_DISCONNECT,
        SYSEVENT_NONE,  // Put it at the end, when using this, it means that no logs will be output (!!!Use with caution!!!)
    }

    public class Dbg 
	{
		static public DEBUGLEVEL debugLevel = DEBUGLEVEL.DEBUG;

#if UNITY_EDITOR
		static Dictionary<string, Profile> _profiles = new Dictionary<string, Profile>();
#endif

		public static string getHead()
		{
			return "[" + DateTime.Now.ToString("yyyy-MM-dd HH:mm:ss fff") + "] ";
		}

		public static void INFO_MSG(object s)
		{
			if (DEBUGLEVEL.INFO >= debugLevel)
				Debug.Log(getHead() + s);
		}

		public static void DEBUG_MSG(object s)
		{
			if (DEBUGLEVEL.DEBUG >= debugLevel)
				Debug.Log(getHead() + s);
		}

		public static void WARNING_MSG(object s)
		{
			if (DEBUGLEVEL.WARNING >= debugLevel)
				Debug.LogWarning(getHead() + s);
		}

		public static void ERROR_MSG(object s)
		{
			if (DEBUGLEVEL.ERROR >= debugLevel)
				Debug.LogError(getHead() + s);
		}

		public static void profileStart(string name)
		{
#if UNITY_EDITOR
			Profile p = null;
			if(!_profiles.TryGetValue(name, out p))
			{
				p = new Profile(name);
				_profiles.Add(name, p);
			}

			p.start();
#endif
		}

		public static void profileEnd(string name)
		{
#if UNITY_EDITOR
			_profiles[name].end();
#endif
		}
		
	}
}
